### ISSUE TITLE
Can you open source some firmware projects

### ISSUE THREAD

**Issue by FantasyGmm** [pebri86/esplay-micro-firmware-collections#2]
I have a lot of firmware does not work, I want to see if I can fix it myself

**Comment by pebri86** [comment]
Hi, what firmware did you mean? All of them are opensource firmware. You can find the source code on github

**Comment by FantasyGmm** [comment]
Ok, I will try to modify it. If I use a newer IDF, such as 4.4, can I generate fw files and boot them by  base firmware?

**Comment by pebri86** [comment]
> Ok, I will try to modify it. If I use a newer IDF, such as 4.4, can I generate fw files and boot them by base firmware?

of course newer IDF will also work with base firmware, but with newer IDF you need to changes some drivers to be compatible.

**Comment by FantasyGmm** [comment]
I modified some of your esplay micro design, added a speaker to the audio output, but is very loud when playing the audio.When I adjust the volume to 6, the volume is still very high, but when I adjust it to 1, it mute. I try to connect a resistor in series at the output end, and the sound becomes smaller, but it sounds a little wrong. Can I modify the software adjustment

**Comment by pebri86** [comment]
> I modified some of your esplay micro design, added a speaker to the audio output, but is very loud when playing the audio.When I adjust the volume to 6, the volume is still very high, but when I adjust it to 1, it mute. I try to connect a resistor in series at the output end, and the sound becomes smaller, but it sounds a little wrong. Can I modify the software adjustment

indeed, you can find the gain value via software, you can learn volume control code on esplay-retro-emulation firmware in the driver code

**Comment by FantasyGmm** [comment]
I also put the latest esp32s3 on the esplay micro. Is it possible to directly modify the basic firmware for adaptation, or rewrite it more convenient

**Comment by pebri86** [comment]
> I also put the latest esp32s3 on the esplay micro. Is it possible to directly modify the basic firmware for adaptation, or rewrite it more convenient

modify to adapt the esp32s3 api is also possible once you figured out how the driver works, but rewrite it will be more convenient as you'll focus on esp32s3 itself rather than figuring out the esp32 api first.
